### PR TITLE
Refine APIs of per-resource slot chain and improve rule slot registering logic in rule managers

### PR DESCRIPTION
**Pull request by louyuting**
### Describe what this PR does / why we need it
Refine code logic related to resource slot chain

### Special notes for reviews

### CODE REVIEW

File: core/base/slot_chain.go [hunk 1]
```diff
@@ -121,6 +133,18 @@ func (sc *SlotChain) AddStatPrepareSlotLast(s StatPrepareSlot) {
 	sc.statPres = append(sc.statPres, s)
 }
 
+func ValidateRuleCheckSlot(sc *SlotChain, s RuleCheckSlot) bool {
```
Comment (sczyh30) [inline core/base/slot_chain.go:136]: Maybe the name could be more concrete?

Comment (louyuting) [reply core/base/slot_chain.go]: rename to ValidateRuleCheckSlotNaming

**Review by sczyh30 — APPROVED** [review]
LGTM

**Comment by sczyh30** [comment]
Thanks!

**Pull request merged**